apps/studio/src/pages/sites/[siteId]/admin.tsx

```diff
@@ -0,0 +1,229 @@
+
+const SUPPORTED_SITE_CONFIG_TYPES = [
+  "config",
+  "theme",
+  "navbar",
+  "footer",
+] as const
+
+const SiteAdminPage: NextPageWithLayout = () => {
+  const toast = useToast()
+  const router = useRouter()
+  const trpcUtils = trpc.useUtils()
+  const { siteId } = useQueryParse(siteAdminSchema)
+  const isUserIsomerAdmin = useIsUserIsomerAdmin()
+
+  if (isUserIsomerAdmin) {
+    toast({
+      title: "You do not have permission to access this page.",
+      status: "error",
+      ...BRIEF_TOAST_SETTINGS,
+    })
+    void router.push(`/sites/${siteId}`)
+  }
+
+  const { mutate, isLoading } = trpc.site.setSiteConfigAdmin.useMutation({
+    onSuccess: async () => {
+      // reset({ config, theme, navbar, footer })
+      await trpcUtils.site.getConfig.invalidate({ id: siteId })
+      await trpcUtils.site.getTheme.invalidate({ id: siteId })
+      await trpcUtils.site.getNavbar.invalidate({ id: siteId })
+      await trpcUtils.site.getFooter.invalidate({ id: siteId })
+      toast({
+        title: "Saved site config!",
+        description: "Check your site in 5-10 minutes to view it live.",
+        status: "success",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+    onError: () => {
+      toast({
+        title: "Error saving site config!",
+        description:
+          "If this persists, please report this issue at support@isomer.gov.sg",
+        status: "error",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+  })
+
+  const [previousConfig] = trpc.site.getConfig.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousTheme] = trpc.site.getTheme.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousNavbar] = trpc.site.getNavbar.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousFooter] = trpc.site.getFooter.useSuspenseQuery({
+    id: siteId,
+  })
+
+  // NOTE: Refining the setNotificationSchema here instead of in site.ts since omit does not work after refine
+  const {
+    register,
+    handleSubmit,
+    formState: { isDirty, errors },
+  } = useZodForm({
+    schema: setSiteConfigAdminSchema
+      .omit({ siteId: true })
+      .refine((data) => !!data.config, {
+        message: "Site config must be present",
+        path: ["config"],
+      })
+      .refine((data) => !!data.theme, {
+        message: "Site theme must be present",
+        path: ["theme"],
+      })
+      .refine((data) => !!data.navbar, {
+        message: "Site navbar must be present",
+        path: ["navbar"],
+      })
+      .refine((data) => !!data.footer, {
+        message: "Site footer must be present",
+        path: ["footer"],
+      }),
+    defaultValues: {
+      config: JSON.stringify(previousConfig, null, 2),
+      theme: JSON.stringify(previousTheme, null, 2),
+      navbar: JSON.stringify(previousNavbar.content, null, 2),
+      footer: JSON.stringify(previousFooter.content, null, 2),
+    },
+  })
+
+  const [nextUrl, setNextUrl] = useState("")
+  const isOpen = !!nextUrl
+
+  useEffect(() => {
```

Comment:
ok, sounds good! this honestly look abit sus but given that it's used on `settings.tsx` , i think it's ok.

i think it might be worth to double-check the function is cleaning up properly; just based off what i see here, we turn off the events if `isDirty` then we `setNextUrl`,  but we only turn on the event when the modal is closed again? wondering if it's possible for our users to bypass the `!isOpen` and for `routeChangeStart` to be off permanently

based off what i know i don't think so, because any closing of the modal will trigger the state change, but just wanted to be sure